zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -101,7 +101,7 @@ public class PrepRequestProcessor extends ZooKeeperCriticalThread implements Req
      */
     private static boolean failCreate = false;
 
-    LinkedBlockingQueue<Request> submittedRequests = new LinkedBlockingQueue<>();
+    BatchedArrayBlockingQueue<Request> submittedRequests = new BatchedArrayBlockingQueue<>(10240);
```

Comment:
You changed the queue from an unbounded queue to a bounded queue of max size 10240, does it have any side effects?

Reply:
We use `put` to take the place of `add.` If the enqueued request is full, the put operation will block.